src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -435,8 +435,9 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
 
         // Try looking for methods and associated items.
         let mut split = path_str.rsplitn(2, "::");
-        // this can be an `unwrap()` because we ensure the link is never empty
-        let (item_str, item_name) = split.next().map(|i| (i, Symbol::intern(i))).unwrap();
+        // NB: the `splitn`'s first element is always defined, even if the delimiter is not present.
```

Comment:
I think the old comment was more clear.

Reply:
Oh wait, I didn't realize the first element would be `Some` if the string is empty. Can you add a `assert_ne!(item_str, "")`, then?